### DISCUSSION TITLE
filelist is showing length 0 . input file  not get assigned with register.

### DISCUSSION THREAD

**Discussion by deepakchandrawanshi1063** [react-hook-form/react-hook-form#6851] [Q&A]
//Publish is MUI icon
//I used both register and control combined on this page. 
//I don't understand why I am getting filelist length as 0 after selecting a file. but file value is get assigned in (BankImg) useState. 
//when I am using control in the place of register i am getting bank_pic value as undefined.

---------------------------------------------------------------------------------------------------
 const { control, register ,formState:{errors}} = useFormContext();
 const [BankImg,setBankImg]=useState('http://www.qsstudy.com/wp-content/uploads/2017/08/Format-of-Bank-Pass-Book.jpg');

 <Grid container component={Box} mt={2}>
          <Grid item xs={6} sm={6} md={4}>
              <>
                <label className="custom-file-upload" style={{
                      border: "1px solid #ccc",
                      display: "inline-block",
                      padding: "6px 12px",
                      margin:"50px 0px 0px 0px",
                      cursor: "pointer",
                      borderRadius: "5%"
                    }}><Publish color="primary"/> Bank Img
                <input
                  type="file"
                  name="bank_pic"
                  id="bank_pic"
                  accept="image/*"
                  {...register("bank_pic")}
                  required="true"
                  style={{ display: "none" }}
                  onChange={(e) => {
                          const size = e.target.files[0].size / 1000;
                          // console.log("size", size);
                          const bankErr=document.getElementById('bankErr');
                          if (size > 500) {
                            bankErr.innerHTML = `upload upto 500 kb image. your size is ${Math.floor(
                              size
                            )} kb`;
                          } else {
                            setBankImg(URL.createObjectURL(e.target.files[0]));
                            bankErr.innerHTML = "";
                          }
                        }}
                />
                <p id="bankErr" style={{ color: "red" }}></p>
                </label>
                </>
          </Grid>

**Comment by JamesWong1999** [comment]
can you share a codesandbox?
